griptape/artifacts/base_artifact.py

```diff
@@ -24,22 +35,7 @@ class BaseArtifact(SerializableMixin, ABC):
         metadata={"serializable": True},
     )
     value: Any = field()
-
-    @classmethod
-    def value_to_bytes(cls, value: Any) -> bytes:
-        if isinstance(value, bytes):
-            return value
-        else:
-            return str(value).encode()
-
-    @classmethod
-    def value_to_dict(cls, value: Any) -> dict:
-        dict_value = value if isinstance(value, dict) else json.loads(value)
-
-        return dict(dict_value.items())
-
-    def to_text(self) -> str:
-        return str(self.value)
+    encoding: str = field(default="utf-8", kw_only=True)
```

Comment:
dont see this in the docstring. do we need this? can it just be passed as a param directly to `to_text`? if not, maybe we can rename it to `text_encoding` or `to_text_encoding`.

Reply:
Just updated the docstring and added a missing implementation. This field already existed in a number of Artifacts, I just bumped to the top since we now have a `to_bytes`.

Reply:
i would say we could just pass `encoding`/`encoding_error_handler` directly to the `to_bytes`/`to_text` methods, and have them default to the same thing as whatever `encode`/`decode` defaults to.